test/join.jl

```diff
@@ -573,25 +573,57 @@ end
     @test innerjoin(name, job, on=:ID, validate=(false, false)) == inner
 
     # Make sure ok with various special values
-    for special in [missing, NaN, 0.0, -0.0]
+    for special in [missing, NaN, -0.0]
         name_w_special = DataFrame(ID = [1, 2, 3, special],
                                    Name = ["John Doe", "Jane Doe", "Joe Blogs", "Maria Tester"])
-        @test innerjoin(name_w_special, job, on=:ID, validate=(true, false)) == inner
+        @test_throws ArgumentError innerjoin(name_w_special, job, on=:ID)
+        @test_throws ArgumentError leftjoin(name_w_special, job, on=:ID)
+        @test_throws ArgumentError rightjoin(name_w_special, job, on=:ID)
+        @test_throws ArgumentError outerjoin(name_w_special, job, on=:ID)
+        @test_throws ArgumentError semijoin(name_w_special, job, on=:ID)
+        @test_throws ArgumentError antijoin(name_w_special, job, on=:ID)
+    end
+
+    for special in [missing, 0.0]
```

Comment:
`0.0` works even without `matchmissing=:equal`, right?

Reply:
right, but I wanted to avoid adding the second set of the same tests. I think it is OK, as `:equal` has no influence on `0.0` handling